pkg/publicapi/middleware/version.go

```diff
@@ -64,7 +76,7 @@ func VersionNotifyLogger(logger *zerolog.Logger, serverVersion semver.Version) e
 			}
 
 			// there is a single version header, attempt to parse it.
-			clientVersion, err := semver.NewVersion(cVersion[0])
+			clientVersion, err := parseVersion(cVersion[0])
```

Comment:
We should instead make a client version with the pre-release tag removed via `clientVersion.SetPrerelease("")`